packages/core/src/components/cms/ViewportObserver.tsx

```diff
@@ -0,0 +1,93 @@
+import type {
+  DetailedHTMLProps,
+  HTMLAttributes,
+  PropsWithChildren,
+} from 'react'
+import { useCallback, useEffect, useRef, useState } from 'react'
+
+// Mobile height to prevent sections outside the viewport from being rendered initially.
+// We are using the Moto G Power device measurement as a reference, as used by PageSpeed Insights.
+const VIEWPORT_SIZE = 823
+
+type ViewportObserverProps = {
+  /**
+   * Identify the store section
+   */
+  name?: string
+  /**
+   * Debug/test purposes: enables visual debugging to identify the visibility of the section.
+   */
+  debug?: boolean
+} & IntersectionObserverInit &
+  DetailedHTMLProps<HTMLAttributes<HTMLElement>, HTMLElement>
```

Comment:
*question*: This** type ` DetailedHTMLProps<HTMLAttributes<HTMLElement>, HTMLElement>` is really necessary?

Reply:
I don't think it is, it was automatically added as suggestion in the editor 🙇 
Do u have any concerns? I'm okay removing it

Reply:
When someone uses this component, the editor will show that we can pass several properties related to `DetailedHTMLProps<HTMLAttributes<HTMLElement>, HTMLElement>`. So it can confuse us.

Reply:
I'll remove it!